extensions/amp-facebook-comments/1.0/component.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Preact from '../../../src/preact';
+import {
+  MessageType,
+  ProxyIframeEmbed,
+} from '../../../src/preact/component/3p-frame';
+import {dashToUnderline} from '../../../src/string';
+import {deserializeMessage} from '../../../src/3p-frame-messaging';
+import {forwardRef} from '../../../src/preact/compat';
+import {tryParseJson} from '../../../src/json';
+import {useCallback, useLayoutEffect, useState} from '../../../src/preact';
+
+/** @const {string} */
+const TYPE = 'facebook';
+const FULL_HEIGHT = '100%';
+const MATCHES_MESSAGING_ORIGIN = () => true;
+const DEFAULT_TITLE = 'Facebook comments';
+
+/**
+ * @param {!FacebookCommentsDef.Props} props
+ * @param {{current: (!FacebookCommentsDef.Api|null)}} ref
+ * @return {PreactDef.Renderable}
+ */
+function FacebookCommentsWithRef(
+  {
+    colorScheme,
+    href,
+    locale: localeProp,
+    numPosts,
+    onReady,
+    orderBy,
+    requestResize,
+    title = DEFAULT_TITLE,
+    ...rest
+  },
+  ref
+) {
+  const [height, setHeight] = useState(FULL_HEIGHT);
+  const messageHandler = useCallback(
+    (event) => {
+      const data = tryParseJson(event.data) ?? deserializeMessage(event.data);
+      if (data['action'] == 'ready') {
```

Comment:
Nit: the singular condition can be added here.

`if (data['action'] == 'ready' && onReady) { onReady() }`

Reply:
Or: `if (data['action'] == 'ready') { onReady?.() }`